### ISSUE TITLE
Snowdrop / Red Hat Runtimes - website information review

### ISSUE THREAD

**Issue by BarDweller** [snowdrop/team#266]
We currently have snowdrop.me, and exist as a 'Red Hat Runtime'. Our presence on the official Red Hat website for Runtimes is inconsistent, and it would appear the main site may need updates, as it refers to Runtimes as only for Java. 

- Firstly.. the Red Hat Runtimes main page : https://www.redhat.com/en/products/runtimes
First mention of Spring (or for that matter Node.js) is after the table listing 'Red Hat Runtimes' in which Quarkus gets its own mention, where as Spring is relegated to being hidden in the undefined "A set of cloud-native runtimes". 
This feels inconsistent, arguably some of the things in the 'Runtimes' table are not actually runtimes at all (eg, Single sign-on? that's a service, so is AMQ, Data Grid etc.) Either the table needs renaming, or we need to list the actual runtimes in in it, and move the related-to-runtimes-but-not-actually-a-runtime-itself to a new column. (Also, why is Red Hat Runtimes in the left table? isn't that what the page is about? and why is AMQ in both?)
Spring only gets a mention in the 'Runtimes and Frameworks' tile, but with no link to further info. 
In the final set of tiles, The 'Cloud-native runtimes' tile says that it is for Java and Javascript based runtimes, this would arguably include Quarkus, yet Quarkus gets it's own tile. 

- Secondly the 'cloud-native runtimes' tile, AND the 'set of cloud-native runtimes' link in the table on the main page lead to https://www.redhat.com/en/technologies/cloud-computing/openshift/application-runtimes/features
Once again the statement is made that red hat runtimes supports a collection of cloud-native runtimes for developing Java and JavaScript. No mention is made of other runtimes.
Paragraphs on this page explain thorntail, vert.x, node.js, but not spring, or quarkus. Odd for a page touting that it covers Java based runtimes. 
Also at the base of this page, the Paragraph explaining the 'set of cloud-native runtimes' links back to this page, which feels a tad pointless. 
The ONLY mention of spring on this page, is part of the JBoss Web Server description right at the end. That feels a bit odd.

- Thirdly, If we follow the link from the 2nd page to https://developers.redhat.com/launch/
We discover the runtimes include .net, node.js, quarkus, snowdrop, some oddball green bird that I have no idea about, wildfly, vert.x and a ufo which I thankfully understand to be openliberty.
This is the first mention of snowdrop in the entire experience. However, note that the 'Learn More' link goes to Spring's documentation, not snowdrop. 
If from this page I follow the steps to create an app (Start button, Deploy an example Application, Select an Example), I'm offered 
Spring Boot, Golang, Node.js, OpenLiberty, Thorntail, and Vert.x as choices. Selecting Spring boot allows me to pick from (Community) and (Red Hat Runtimes) versions of Spring. No mention is made of the implications of that choice, no links are made to explain that 'Red Hat Runtimes' in this case is supplied by Snowdrop. 

- In general....
No links at any stage have associated the Spring Red Hat Runtime with Snowdrop, and no links have explained why anyone would want to use the Red Hat version over the community version. 
The only place I could find an association was via the 'Try it' link on the 2nd page, which amusingly leads to a DIFFERENT site than the 'Try it now' link on the same page.. (yes, really). The Try it link leads to this URL https://developers.redhat.com/products/rhoar/download?extIdCarryOver=true&sc_cid=701f2000001OH7TAAW  which offers you Spring Boot and Thorntail (no node.js, no vert.x etc.. all gone!) .. And here is a box that implies Snowdrop is an upstream of a runtime, but it doesn't say which, so you'd need to know that to follow the link if you were spring inclined. 
It's only when you get over to the support portal, which is different from the main site, do you get a slightly different view - https://access.redhat.com/products/red-hat-runtimes/
Here we find Spring Boot mentioned, with a dedicated link for 'Red Hat support for Spring Boot' (note the leading paragraph omits Quarkus, but it does also have it's own link below). Personally I'd still want to reorganise the portfolio to group actual runtimes apart from services like sso. 
The Red Hat Spring Boot info link here is almost useful ;) https://access.redhat.com/products/spring-boot although this is stuff aimed at people who already bought it.. so it's a bit late to be explaining stuff here. 

- Version notes.. 
Across all the different sites, the version offered appears to be 2.2.6 (eg, Try It -> links to 2.2.6, Try it Now -> offers 2.2.6, the product docs site (https://access.redhat.com/documentation/en-us/red_hat_support_for_spring_boot/2.2/) expressly says 'For use with Spring Boot 2.2.6)

- 2.3 doesn't even appear to be an option, although it's been out since May 2020.
- 2.1 is still offered at the docs site, even tho that goes eol at the end of Oct 2020 (this month https://spring.io/blog/2019/12/10/spring-boot-2-1-x-eol-november-1st-2020 )
- 2.2 will go EOL in October 2022, so there's a while for that to live yet. 
- (as per https://github.com/spring-projects/spring-boot/wiki/Supported-Versions)

**Comment by cmoulliard** [comment]
> We currently have snowdrop.me, and exist as a 'Red Hat Runtime'.

What do you want to say here ? Can you rephrase please ?

**Comment by cmoulliard** [comment]
> Our presence on the official Red Hat website for Runtimes is inconsistent, and it would appear the main site may need updates, as it refers to Runtimes as only for Java.

As you cover different aspects within this ticket, can you then better organize the content and use sub-titles please

Example:

```
## Red Hat Customer portal and Runtimes

### Lack of reference of Spring Boot Runtime
...
### Inconsistent versioning
```

**Comment by cmoulliard** [comment]
> * (Also, why is Red Hat Runtimes in the left table? isn't that what the page is about? and why is AMQ in both?)

This is because this page lists the 3 subscriptions that Red Hat proposes for the customers: Integration, Runtimes and Automation. If you buy the subscription of integration, then you will also got the runtimes (= support)

**Comment by cmoulliard** [comment]
> * 2.3 doesn't even appear to be an option, although it's been out since May 2020.
> *

2.3 will only be added when it will be released. ER1 build has been created yesterday and Libor - QE team is testing it ;-)

**Comment by BarDweller** [comment]
> > We currently have snowdrop.me, and exist as a 'Red Hat Runtime'.
> 
> What do you want to say here ? Can you rephrase please ?

If I understand this correctly, Snowdrop is the upstream for the Spring Boot Red Hat Runtime. It's either that, or it IS the Spring Boot Red Hat Runtime. I'm trying to illustrate there are 2 facets here of the same thing, that we seem to use interchangeably, but yet never explain the relationship between, or define their boundaries of, via the pages.

**Comment by BarDweller** [comment]
> > Our presence on the official Red Hat website for Runtimes is inconsistent, and it would appear the main site may need updates, as it refers to Runtimes as only for Java.
> 
> As you cover different aspects within this ticket, can you then better organize the content and use sub-titles please
> 
> Example:
> 
> ```
> ## Red Hat Customer portal and Runtimes
> 
> ### Lack of reference of Spring Boot Runtime
> ...
> ### Inconsistent versioning
> ```

No probs.. was more focused on getting the words & thoughts captured.. will revisit the layout.

**Comment by BarDweller** [comment]
> > * 2.3 doesn't even appear to be an option, although it's been out since May 2020.
> 
> 2.3 will only be added when it will be released. ER1 build has been created yesterday and Libor - QE team is testing it ;-)

2.3.0 (community) has been available since May, yet the 'launch' thing that offers community versions still only offers 2.2.6. 
The comment was related to the same root cause tho, that 2.2.6 seems to be maximum version referred to across the site, even though 2.2.6 was released back in March, and the 2.2.x line is currently at 2.2.10 ? Looking at https://mvnrepository.com/artifact/dev.snowdrop/snowdrop-dependencies we have released a 2.2.10 (although it's not clear where 2.2.7/2.2.8/2.2.9 went). Either way, you'd expect all the various parts currently saying 2.2.6 to have updated.

**Comment by cmoulliard** [comment]
> If I understand this correctly, Snowdrop is the upstream for the Spring Boot Red Hat Runtime.

Yes. Snowdrop is the name of our team, github domain, DNS name `snowdrop.me | snowdrop.dev`; corresponds also to our upstream projects and where we host projects that we build specifically for Red Hat Runtime Spring Boot `Eg: SpringBoot with Vert.x as Reactive engine`

**Comment by cmoulliard** [comment]
> although it's not clear where 2.2.7/2.2.8/2.2.9 went

They were not released as currently we cannot productize everything that Pivotal is releasing due to time, resources constraints. Question: Should we productize and release them ?

**Comment by cmoulliard** [comment]
I propose that you move this task to the backlog (as we depend on the decision about what we will implement to support / generate a project) and runtime discussions @BarDweller

**Comment by BarDweller** [comment]
Agreed.
